fix: Suppress diff for network policy in user resources (#3657)
snowflakedb/terraform-provider-snowflake: MIGRATION_GUIDE.md

Reference: [#3639](https://github.com/snowflakedb/terraform-provider-snowflake/issues/3639).

> ### *(bugfix)* Added diff suppression to `network_policy` to all user resources

Reviewer: ```suggestion
### *(bugfix)* Fixed permadiff issue with `network_policy` in user resources
```

Author: Done.
